### PR TITLE
2492: Update developer onboarding ticket

### LINKED ISSUE #2492: Update the developer onboarding template and document the HW/SW needs of an engineer on the team
### Issue description

To enable the agency to better meet our needs, we need to document the software (and general permissions) needed that would allow a federal engineer to get our application up and running locally, enable them to develop it, and deploy it to our environment(s). Most installs are currently documented in our onboarding ticket, but we should make sure that encompasses all of them.

This should be fast to do and less than a day's worth of time. We need to have a solid initial version of the HW/SW doc by COB July 30.

### Acceptance criteria
- [ ] Update the [developer onboarding issue template](https://github.com/cisagov/manage.get.gov/blob/main/.github/ISSUE_TEMPLATE/developer-onboarding.md) to include all softwares needed to get the application running on any machine. This does not need to include language for requesting access to certain things (as hopefully we will handle that)
  - [ ] include some things that are currently known as lacking: 
    - [ ] install docker desktop
    - [ ] have a device capable of installing python files (especially with pip install)
    - [ ] have a device capable of installing node packages (especially with npm install) 
- [ ] [Document](https://docs.google.com/document/d/1ddl_PUA-l2pPLovrLhkcp3CQJjaKmMiAO6n9gHY0N6k/edit?usp=sharing) the software/hardware requirements for a .gov engineer. 
  - [ ] From a software standpoint, this should largely mirror the developer onboarding template. Include other needs, like an IDE or other local tools needed to write, build, or troubleshoot. This may also include permissions or abilities (like "have a device capable of installing python files (especially with pip install)")
   - [ ] Any minimum hardware requirements (RAM, storage)
- [ ] Go over both docs in eng huddle, or in a group discussion in slack within the week to get feedback on anything that could be missing. 

### Additional context
- The second document linked should be written as if telling a third party about our system and software needs without linking to any of our other documentation. This is a one off task.
- We should anticipate that the hardware eventually shipped to users runs Windows.
- Go through the [developer onboarding guide](https://docs.google.com/document/d/1ukbpW4LSqkb_CCt8LWfpehP03qqfyYfvK3Fl21NaEq8/edit#heading=h.n4rh22vwk68n) and remove any software installation steps and add them to the SWHD doc and the developer onboarding template. 

### Links to other issues

- Replaces #2325

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #2492 

## Changes

- Adds required and recommended software installations on DHS device.
- Adds known issues to setting up developer environment on DHS device.

## Context for reviewers

Changes to be discussed in 7/29 engineering huddle. Also interested in opening discussion to explore ways we can organize onboarding to be easier to follow. Considerations include DHS device setup, public/private onboarding info, and the number of docs/links onboarding is spread across.

This PR would ideally deprecate the need for the [Developing on a CISA laptop doc](https://docs.google.com/document/d/1ZyaaTWOEgVEHiHarNuv97CnYfUFe-4pfbuQm4ocLkn4/edit).

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/developer-onboarding.md [hunk 1]
```diff
@@ -12,15 +12,29 @@ assignees: abroddrick
 - Onboardee: _GH handle of person being onboarded_
 - Onboarder: _GH handle of onboard buddy_
 
-## Installation
+## Installation
```
[the remark this answers is not in the document]
Comment (erinysong) [reply .github/ISSUE_TEMPLATE/developer-onboarding.md]: Good catch - thank you for getting that!

File: .github/ISSUE_TEMPLATE/developer-onboarding.md [hunk 2]
```diff
@@ -12,15 +12,29 @@ assignees: abroddrick
...
   - Alternatively, for Windows, [consider using chocolately](https://community.chocolatey.org/packages/cloudfoundry-cli/7.2.0)
-- [ ] Make sure you have `gpg` >2.1.7. Run `gpg --version` to check. If not, [install gnupg](https://formulae.brew.sh/formula/gnupg)
-  - Alternatively, you can skip this step and [use ssh keys](#setting-up-commit-signing-with-ssh) instead
-- [ ] Install the [Github CLI](https://cli.github.com/)
+- [ ] [GPG](https://gnupg.org/download/)
+  - This may not work on DHS devices. Instead, you can [use ssh keys](#setting-up-commit-signing-with-ssh) instead
+- [ ] *Docker Community Edition
+- [ ] *Git
+- [ ] *VSCode (our preferred editor)
+- [ ] *Github Desktop
```
[the remark this answers is not in the document]
Comment (erinysong) [reply .github/ISSUE_TEMPLATE/developer-onboarding.md]: Unrelated, but thoughts on moving this to recommended software? I don't think Github CLI/Desktop are necessary for development and more a quality of life option

File: .github/ISSUE_TEMPLATE/developer-onboarding.md [hunk 3]
```diff
@@ -12,15 +12,29 @@ assignees: abroddrick
...
+
+The following tools are optional, but also can be requested through the DHS IT portal:
```
[the remark this answers is not in the document]
Comment (Matt-Spence) [reply .github/ISSUE_TEMPLATE/developer-onboarding.md:29]: Good suggestion.

File: .github/ISSUE_TEMPLATE/developer-onboarding.md [hunk 4]
```diff
@@ -12,15 +12,29 @@ assignees: abroddrick
...
+There are several tools we use locally that you will need to have. 
+
+- [ ] [Cloudfoundry CLI](https://docs.cloudfoundry.org/cf-cli/install-go-cli.html#pkg-mac) Note: If you are on Windows the cli will be under `cf8` or `cf7` depending on which version you install.
   - If you are using Windows, installation information can be found [here](https://github.com/cloudfoundry/cli/wiki/V8-CLI-Installation-Guide#installers-and-compressed-binaries)
   - Alternatively, for Windows, [consider using chocolately](https://community.chocolatey.org/packages/cloudfoundry-cli/7.2.0)
-- [ ] Make sure you have `gpg` >2.1.7. Run `gpg --version` to check. If not, [install gnupg](https://formulae.brew.sh/formula/gnupg)
-  - Alternatively, you can skip this step and [use ssh keys](#setting-up-commit-signing-with-ssh) instead
-- [ ] Install the [Github CLI](https://cli.github.com/)
+- [ ] [GPG](https://gnupg.org/download/)
+  - This may not work on DHS devices. Instead, you can [use ssh keys](#setting-up-commit-signing-with-ssh) instead
```
[the remark this answers is not in the document]
Comment (Matt-Spence) [reply .github/ISSUE_TEMPLATE/developer-onboarding.md:23]: That was not intended to be removed, good catch. The weird wording I think was the result of starting and stopping in the middle of typing, another good catch.

File: .github/ISSUE_TEMPLATE/developer-onboarding.md [hunk 5]
```diff
@@ -14,13 +14,29 @@ assignees: abroddrick
...
-- [ ] Make sure you have `gpg` >2.1.7. Run `gpg --version` to check. If not, [install gnupg](https://formulae.brew.sh/formula/gnupg)
-  - Alternatively, you can skip this step and [use ssh keys](#setting-up-commit-signing-with-ssh) instead
-- [ ] Install the [Github CLI](https://cli.github.com/)
+- [ ] [GPG](https://gnupg.org/download/)
+  - Make sure you have `gpg` >2.1.7. Run `gpg --version` to check. If not, [install gnupg](https://formulae.brew.sh/formula/gnupg)
+  - This may not work on DHS devices. Alternatively, you can [use ssh keys](#setting-up-commit-signing-with-ssh) instead.
+- [ ] Docker Community Edition*
+- [ ] Git*
+- [ ] VSCode (our preferred editor)*
+- [ ] Github Desktop*
```
[the remark this answers is not in the document]
Comment (Matt-Spence) [reply .github/ISSUE_TEMPLATE/developer-onboarding.md:28]: Good point, what we really want is the github cli for running some of our scripts.

File: .github/ISSUE_TEMPLATE/developer-onboarding.md [hunk 6]
```diff
@@ -14,13 +14,29 @@ assignees: abroddrick
...
+
+The following tools are optional  but recommended. For DHS devices, these can be requested through the DHS IT portal:
+- [ ] Slack Desktop App**
+- [ ] Python 3.10*
+- [ ] NodeJS (latest version available)*
+- [ ] Putty*
+- [ ] Windows Subsystem for Linux*
+
+\* Must be requested through DHS IT portal on DHS devices
```
[the remark this answers is not in the document]
Comment (Matt-Spence) [reply .github/ISSUE_TEMPLATE/developer-onboarding.md:37]: Yep

**Pull request merged**